src/coord/src/coord.rs

```diff
@@ -3565,7 +3567,12 @@ where
         let values = sql::plan::plan_copy_from(&session.pcx(), &catalog, id, columns, rows)?;
 
         let constants = self
-            .prep_relation_expr(values.lower(), ExprPrepStyle::Write)?
```

Comment:
Is it a problem to keep this as .lower? COPY is always only constants, and just two weeks ago we added code to not run the optimizer for it: https://github.com/MaterializeInc/materialize/pull/8694.

Reply:
I think this is fine for now, as `HirRelationExpr::lower()` is going to be around even after we introduce QGM #8073. Eventually though, we might want to remove it, and I guess the proper way to achieve the same bypassing after that will be to call `values.optimize` with an `OptimizerConfig` that indicates that `HIR` ⇒ `MIR` shortcut for constants should be applied.

Reply:
Done.